### ISSUE TITLE
Random crashes while exploring the world (1.12.2)

### ISSUE THREAD

**Issue by CrazyDev05** [ShetiPhian/Localizations-and-Issues#394]
**Describe the bug**
Random crashes while exploring the world

**Additional context**
Minecraft Version: 1.12.2
Forge Version: 14.23.5.2860
ShetiPhian Core Version: 3.5.9
My Mod(s) Name & Version: 
- too many, here is the pack:
https://ann.nl.tab.digital/s/gpD6TmNyjSFdmcd

Are other mods present?
If yes:
- Does everything work without my mods?
  Yes
- Have you tested with only my mods?
   No, but my pack is based on ATM3 and ATM3 works just fine.

PasteBin or GitHub Gists links
crash-report = [crash-2023-01-24_20.42.37-server (1).txt](https://github.com/ShetiPhian/Localizations-and-Issues/files/10493463/crash-2023-01-24_20.42.37-server.1.txt)
latest.log = [2023-01-24-7.log.gz](https://github.com/ShetiPhian/Localizations-and-Issues/files/10493478/2023-01-24-7.log.gz)
debug.log = [debug-1.log.gz](https://github.com/ShetiPhian/Localizations-and-Issues/files/10493475/debug-1.log.gz)

**Comment by CrazyDev05** [comment]
[crash-2023-01-24_20.59.45-server.txt](https://github.com/ShetiPhian/Localizations-and-Issues/files/10493515/crash-2023-01-24_20.59.45-server.txt)
again, but this time with a player

**Comment by ShetiPhian** [comment]
Support for 1.12.2 was dropped years ago.


I looked up the line crashing in my version control and it is checking the entries in the OreDictionary.
So that crash indicates some mod is incorrectly adding null to the dictionary.


While it would be best to find the mod adding null as it can cause other issues, in a large pack it can be hard to pin down.
In the configuration file you can change the doodad whitelists and blacklists so there is no "ore:" entries, that way the OreDictionary is not scanned. (you'll need to define each drop though)
You could also just disable the doodads via configs.

**Comment by Kasia1403** [comment]
What are doodads ? I disabled them.

**Comment by CrazyDev05** [comment]
@ShetiPhian
would I be allowed to change the code to print out the broken ore dictionary entry?

**Comment by ShetiPhian** [comment]
By that point the entry is null so the only info to gain is what list it is
in. (we know it's one of the blacklisted ones)
If your patching the code a null check would be more helpful as it would
stop the crash.
Hooking the OreDictionary itself would be more useful. You could watch for
null to be registered, block it and the stack trace would point out what
was doing it.
But if nothing registers null then the value is being changed later, so
hooking the fetch method and validating the return would fix it but still
wouldn't know the cause.

**Comment by CrazyDev05** [comment]
could you send me the source code or the class, please, because decompiling could produce other problems?

**Comment by ShetiPhian** [comment]
I figured you'd be making a patcher mod that used ASM Bytecode or Mixins (if available for 1.12)

[The source](https://gist.github.com/ShetiPhian/29676bb8544bc2141f5dd3dcae2c56d5) is written on an old build environment that can no longer be setup or compile, and is using mappings that no longer exist.

Using gradle sou should be able convert to the latest mappings. (works in fabric 1.19, never tested on forge)
Create a 1.12.2 gradle project
add to the build.gradle file:
```
repositories {
    maven { url "https://www.cursemaven.com" }
}

dependencies {
    // Fabric uses modImplementation() instead
    compile fg.deobf("curse.maven:shetiphiancore-71738:2661618")
    compile fg.deobf("curse.maven:terraqueous-244870:2710209")
}
```

The two mods will be added to the external libraries and be remapped to the current mappings

From there you create the package shetiphian.terraqueous.common.block and the BlockDoodad class
Copy the code from the library, make your changes and build
Lastly copy the class from the jar you just built into Terraqueous replacing the original. (the jars can be opened as zip files)
